homework/unique_ptr/UniquePtr.hpp

```diff
@@ -0,0 +1,73 @@
+// Jest klasą szablonową
+// Trzyma wskaźnik do zarządzanego obiektu
+// Konstruktor kopiuje wskaźnik
+// Destruktor zwalnia pamięć
+// Kopiowanie jest niedozwolone
+// Przenoszenie jest dozwolone i oznacza:
+//     Skopiowanie wskaźnika z obiektu źródłowego
+//     Ustawienie wskaźnika w obiekcie źródłowym na nullptr
+// Wymagane metody: operator*(), operator->(), get(), release(), reset()
+
+template<typename T>
+class UniquePtr {
+public:
+    UniquePtr() = default;
+
+    explicit UniquePtr(T* ptr) {
+        ptr_ = ptr;
+    }
+
+    UniquePtr(const UniquePtr& copy) = delete;
+
+    UniquePtr(UniquePtr&& move) {
+        ptr_ = move.ptr_;
+        move.ptr_ = nullptr;
+    }
+
+    ~UniquePtr() {
+        if (ptr_) {
+            delete ptr_;
+        }
+    }
+    
+    UniquePtr& operator=(const UniquePtr& copy) = delete;
+    UniquePtr& operator=(UniquePtr&& move) {
+        if (&move != this) {
+            delete ptr_;
+            ptr_ = move.ptr_;
+            move.ptr_ = nullptr;
+        }
+        return *this;
+    }
+
+    T& operator*() const {
+        if (!ptr_) {
+            throw std::runtime_error("wrong operation poiter = nullptr");
+        }
+        return *ptr_;
+    }
+
+    T* operator->() const {
+        return ptr_;
+    }
+
+    T* get() const {
+        return ptr_;
+    }
+
+    T* release() {
+        T* temp { ptr_ };
+        ptr_ = nullptr;
+        return temp;
+    }
+
+    void reset(T* data) {
```

Comment:
```suggestion
    void reset(T* data = nullptr) {
```
It should allow a usage like that:
```cpp
ptr.reset();
```

Please add a proper test for this case as well.